storage/innobase/row/row0ins.cc

```diff
@@ -3559,6 +3561,7 @@ row_ins_index_entry_step(
 					   node->row);
 
 	if (err != DB_SUCCESS) {
+		thr_get_trx(thr)->error_info = node->index;
```

Comment:
This had better be done in `row_ins_index_entry_set_vals()` right before the error code `DB_CANT_CREATE_GEOMETRY_OBJECT` is being returned. It seems to me that the function parameters had better be `node, thr` so that `row_ins_index_entry_step()` pass its parameters as is to that function.